.github/workflows/test_linux_cuda.yaml

```diff
@@ -0,0 +1,68 @@
+
+name: Unit-tests on Linux GPU
+
+on:
+  pull_request:
+  push:
+    branches:
+      - nightly
+      - main
+      - release/*
+  workflow_dispatch:
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        python_version: ["3.8", "3.9", "3.10"]
+        cuda_arch_version: ["12.4"]
+      fail-fast: false
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    with:
+      runner: linux.g5.4xlarge.nvidia.gpu
+      repository: pytorch/torchcodec
+      gpu-arch-type: cuda
+      gpu-arch-version: ${{ matrix.cuda_arch_version }}
+      timeout: 120
+
+      script: |
+        nvidia-smi
+        ls -l
+        pwd
+        conda create --yes --name test
+        conda activate test
+        conda install --yes pip cmake pkg-config nasm
+
+        # This fails because conda reactivate fails inside this env
+        # conda install --yes conda-forge::compilers
+
+        pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu124
+        conda install --yes nvidia::libnpp
+
+        # Build and install FFMPEG from source with CUDA enabled.
+        # The one on conda doesn't have CUDA enabled.
+        # Sub-step: install nvidia headers. Reference this link for details:
+        # https://docs.nvidia.com/video-technologies/video-codec-sdk/12.1/ffmpeg-with-nvidia-gpu/index.html
+        git clone https://git.videolan.org/git/ffmpeg/nv-codec-headers.git
+
+        pushd nv-codec-headers
+        make PREFIX=$CONDA_PREFIX -j install
+        popd
+
+        # Now build FFMPEG from source with CUDA enabled.
+        git clone https://git.ffmpeg.org/ffmpeg.git ffmpeg/
+        pushd ffmpeg
+        git checkout origin/release/6.1
+        which pkg-config
+        pkg-config --list-all
+        ./configure --prefix=$CONDA_PREFIX --enable-nonfree --enable-cuda-nvcc --disable-static --enable-shared --optflags=-fno-omit-frame-pointer --disable-stripping --enable-cuvid
+        make -j install
+        popd
+
+        CMAKE_BUILD_PARALLEL_LEVEL=8 CXXFLAGS="" LDFLAGS="-Wl,--allow-shlib-undefined -Wl,-rpath,$CONDA_PREFIX/lib -Wl,-rpath-link,$CONDA_PREFIX/lib -L$CONDA_PREFIX/lib" CMAKE_BUILD_TYPE=Release ENABLE_CUDA=1 ENABLE_NVTX=1 pip install -e . --no-build-isolation -vv --debug
+        pip install pytest
+
+        # We skip certain tests because they are not relevant to GPU decoding and they always fail with
+        # a custom FFMPEG build.
+        pytest -k "not (test_get_metadata or get_ffmpeg_version)"
```

Comment:
So, fun fact: it's currently hard to tell whether the GPU tests are being ran at all, because they're protected within an "if" block. I'll share pointers on how to address this offline

Reply:
This is accurate and a real problem. Should I address this in a different diff?